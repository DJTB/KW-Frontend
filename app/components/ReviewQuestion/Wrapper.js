import styled from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 100%;
  text-align: center;
`;

export default Wrapper;
