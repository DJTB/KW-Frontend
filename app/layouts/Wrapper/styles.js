import React from 'react';
import styled, { css } from 'styled-components';

import { centerByPadding, centerByMargin } from 'shared/styles/layout';

const style = css`
  ${({ fullWidth, fullWidthBg }) => {
    if (fullWidth) {
      return 'width: 100%;';
    }
    if (fullWidthBg) {
      return `
        position: relative;
        ${centerByPadding}
      `;
    }
    return centerByMargin;
  }}
`;

export const StyledWrapper = styled(({
  tag,
  children,
  fullWidth,
  fullWidthBg,
  ...props
}) => React.createElement(tag, props, children))`${style}`;
