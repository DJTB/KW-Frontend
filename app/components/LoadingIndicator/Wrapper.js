import styled from 'styled-components';

import { gutter } from 'shared/styles/layout';

const Wrapper = styled.div`
  ${gutter({ type: 'outer', prop: 'margin', direction: 'vertical' })}
  position: relative;
  width: 2rem;
  height: 2rem;
  margin-left: auto;
  margin-right: auto;
  color: ${({ color }) => color}
`;

export default Wrapper;
