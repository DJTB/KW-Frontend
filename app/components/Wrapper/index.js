import React, { PropTypes } from 'react';
import styled from 'styled-components';
import { wrapperGutter } from 'shared/styles/layout';

const StyledDiv = styled.div`
  ${(props) => props.padding ? wrapperGutter : ''}
`;

const Wrapper = ({ children, withPadding }) => (
  <StyledDiv padding={withPadding}>
    {children}
  </StyledDiv>
  );

Wrapper.propTypes = {
  withPadding: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.node,
  ]).isRequired,

};

Wrapper.defaultProps = {
  withPadding: true,
};

export default Wrapper;
