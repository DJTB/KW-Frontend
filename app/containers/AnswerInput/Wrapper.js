import styled from 'styled-components';
import { shadowBox } from 'shared/styles/shadows';

export default styled.div`
  position: relative;
  color: currentColor;
  ${shadowBox}
  transition: background-color 150ms ease-in-out;
`;
