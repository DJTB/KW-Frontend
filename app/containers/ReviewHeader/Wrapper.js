import styled from 'styled-components';
import { white } from 'shared/styles/colors';

const Wrapper = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  color: ${white}
`;
export default Wrapper;
