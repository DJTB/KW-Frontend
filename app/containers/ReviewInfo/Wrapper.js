import styled from 'styled-components';
import { whiteLight } from 'shared/styles/colors';


const Wrapper = styled.div`
  display: table-row;
  width: 100%;
  height: 100%;
  background-color: ${whiteLight};
`;

export default Wrapper;
