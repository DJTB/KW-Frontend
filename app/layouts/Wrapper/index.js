import React from 'react';
import PropTypes from 'prop-types';

import { StyledWrapper } from './styles';

Wrapper.propTypes = {
  fullWidth: PropTypes.bool,
  fullWidthBg: PropTypes.bool,
  tag: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.node,
  ]).isRequired,
};

Wrapper.defaultProps = {
  tag: 'div',
  fullWidth: false,
  fullWidthBg: false,
};

function Wrapper({ tag, children, fullWidth, fullWidthBg }) {
  return (
    <StyledWrapper
      tag={tag}
      fullWidth={fullWidth}
      fullWidthBg={fullWidthBg}
    >
      {children}
    </StyledWrapper>
  );
}

export default Wrapper;
