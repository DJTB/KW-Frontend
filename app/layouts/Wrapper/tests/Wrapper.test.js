import React from 'react';
import { shallow } from 'enzyme';
import Wrapper from '../index';

describe('<Wrapper />', () => {
  it('should match baseline snapshot', () => {
    const renderedComponent = shallow(
      <Wrapper>text</Wrapper>
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
