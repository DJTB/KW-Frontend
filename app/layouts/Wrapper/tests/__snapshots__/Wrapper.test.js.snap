// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> should match baseline snapshot 1`] = `
<Wrapper
  fullWidth={false}
  fullWidthBg={false}
  tag="div"
>
  <Styled(Component)
    fullWidth={false}
    fullWidthBg={false}
    tag="div"
  >
    <Component
      className="xrKBM"
      fullWidth={false}
      fullWidthBg={false}
      tag="div"
    >
      <div
        className="xrKBM"
      >
        text
      </div>
    </Component>
  </Styled(Component)>
</Wrapper>
`;
